src/file-utils/edfi_lms_file_utils/file_reader.py

```diff
@@ -18,17 +18,42 @@ def _default() -> pd.DataFrame:
     return pd.DataFrame()
 
 
-def _read_csv(file: str, nrows: Union[int, None] = None) -> pd.DataFrame:
+def _read_csv(
+    file: str,
+    nrows: Optional[int] = None,
+    data_types: Dict[str, str] = dict(),
+    date_cols: List[str] = list(),
+) -> pd.DataFrame:
     logger.info(f"Reading file: {file}")
     if file:
+
+        dates = [
+            "SourceCreateDate",
+            "SourceLastModifiedDate",
+            "CreateDate",
+            "LastModifiedDate",
+            *date_cols,
+        ]
+
+        dtype = {
+            **data_types,
+            "SourceSystemIdentifier": "string",
+            "SourceSystem": "string"
+        }
```

Comment:
```suggestion
        dtype = {
            "SourceSystemIdentifier": "string",
            "SourceSystem": "string",
            **data_types,
        }
```
In case of having to override types for sourcesystemidentifier, you can use the same data_types parameter to do so if you put it on this way.

Reply:
That's a nice idea, will take it.